sw/host/opentitanlib/src/util/openocd.rs

```diff
@@ -292,6 +292,60 @@ impl OpenOcdServer {
             }
         }
     }
+
+    // Convert a RISC-V register name into a name that can be used
+    // with read_register32.
+    fn riscv_reg_name(&self, reg: &RiscvReg) -> &'static str {
+        match reg {
+            RiscvReg::GprByName(gpr) => match gpr {
+                RiscvGpr::GP => "gp",
+                RiscvGpr::SP => "sp",
+            },
+            RiscvReg::CsrByName(gpr) => match gpr {
+                RiscvCsr::DPC => "dpc",
+                RiscvCsr::PMPCFG0 => "pmpcfg0",
+                RiscvCsr::PMPCFG1 => "pmpcfg1",
+                RiscvCsr::PMPCFG2 => "pmpcfg2",
+                RiscvCsr::PMPCFG3 => "pmpcfg3",
+                RiscvCsr::PMPADDR0 => "pmpaddr0",
+                RiscvCsr::PMPADDR15 => "pmpaddr15",
+            },
+        }
+    }
+
+    /// Read a register: this function does not attempt to translate the
+    /// name or number of the register. If force is set, bypass OpenOCD's
+    /// register cache.
+    fn read_register<T: ParseInt>(&self, reg_name: &str, force: bool) -> Result<T> {
+        let cmd = format!(
+            "get_reg {} {{ {} }}",
+            if force { "-force" } else { "" },
+            reg_name,
+        );
+        let response = self.send_tcl_cmd(cmd.as_str())?;
+        // the expected output format is 'reg_name 0xabcdef', e.g 'pc 0x10009858'
+        let (out_reg_name, value) = response
+            .trim()
+            .split_once(' ')
+            .context("expected response of the form 'reg value', got '{response}'")?;
+        ensure!(
+            out_reg_name == reg_name,
+            "OpenOCD returned the value for register '{out_reg_name}' instead of '{reg_name}"
```

Comment:
do you want this to be a specific JTAG error?

Reply:
I don't have a strong opinion. We already have a lot of generic openocd error. Maybe we could create a `UnexpectedOpenOCDResponse` error?